### ISSUE TITLE
eu.europa.esig.dss.DSSException: Unable to parse file '/tmp/dss-cache-tsl/PT.xml' : Cannot add overlapping item

### ISSUE THREAD

**Issue by raigoinabox** [open-eid/digidoc4j#51]
I am getting an exception in my logs whenever I try to sign a BDOC container:

```java
java.util.concurrent.ExecutionException: eu.europa.esig.dss.DSSException: Unable to parse file '/tmp/dss-cache-tsl/PT.xml' : Cannot add overlapping item
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:188)
	at eu.europa.esig.dss.tsl.service.TSLValidationJob.analyzeCountryPointers(TSLValidationJob.java:411)
	at eu.europa.esig.dss.tsl.service.TSLValidationJob.refresh(TSLValidationJob.java:260)
	at org.digidoc4j.impl.asic.tsl.LazyTslCertificateSource.refreshTsl(LazyTslCertificateSource.java:127)
	at org.digidoc4j.impl.asic.tsl.LazyTslCertificateSource.initTsl(LazyTslCertificateSource.java:119)
	at org.digidoc4j.impl.asic.tsl.LazyTslCertificateSource.refreshIfCacheExpired(LazyTslCertificateSource.java:105)
	at org.digidoc4j.impl.asic.tsl.ClonedTslCertificateSource.initializeClonedTsl(ClonedTslCertificateSource.java:56)
	at org.digidoc4j.impl.asic.tsl.ClonedTslCertificateSource.getCertificateSource(ClonedTslCertificateSource.java:49)
	at org.digidoc4j.impl.asic.tsl.ClonedTslCertificateSource.getCertificatePool(ClonedTslCertificateSource.java:75)
	at org.digidoc4j.impl.asic.tsl.LazyCertificatePool.getCertificatePool(LazyCertificatePool.java:86)
	at org.digidoc4j.impl.asic.tsl.LazyCertificatePool.getInstance(LazyCertificatePool.java:61)
	at eu.europa.esig.dss.validation.SignatureValidationContext.initialize(SignatureValidationContext.java:131)
	at eu.europa.esig.dss.validation.DefaultAdvancedSignature.getSignatureValidationContext(DefaultAdvancedSignature.java:204)
	at eu.europa.esig.dss.xades.signature.XAdESLevelBaselineLT.extendSignatureTag(XAdESLevelBaselineLT.java:80)
	at eu.europa.esig.dss.xades.signature.XAdESLevelBaselineT.extendSignatures(XAdESLevelBaselineT.java:145)
	at eu.europa.esig.dss.xades.signature.XAdESLevelBaselineT.extendSignatures(XAdESLevelBaselineT.java:78)
	at eu.europa.esig.dss.xades.signature.XAdESService.signDocument(XAdESService.java:126)
	at eu.europa.esig.dss.xades.signature.XAdESService.signDocument(XAdESService.java:148)
	at org.digidoc4j.impl.asic.xades.XadesSigningDssFacade.signDocument(XadesSigningDssFacade.java:119)
	at org.digidoc4j.impl.asic.AsicSignatureBuilder.finalizeSignature(AsicSignatureBuilder.java:119)
	at org.digidoc4j.impl.asic.AsicSignatureBuilder.invokeSigningProcess(AsicSignatureBuilder.java:81)
	at org.digidoc4j.SignatureBuilder.invokeSigning(SignatureBuilder.java:167)
	at *snip* *snip*
Caused by: eu.europa.esig.dss.DSSException: Unable to parse file '/tmp/dss-cache-tsl/PT.xml' : Cannot add overlapping item
	at eu.europa.esig.dss.tsl.service.TSLParser.call(TSLParser.java:139)
	at eu.europa.esig.dss.tsl.service.TSLParser.call(TSLParser.java:105)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	... 1 more
Caused by: java.lang.IllegalArgumentException: Cannot add overlapping item
	at eu.europa.esig.dss.util.MutableTimeDependentValues.addOldest(MutableTimeDependentValues.java:34)
	at eu.europa.esig.dss.tsl.service.TSLParser.getStatusHistory(TSLParser.java:384)
	at eu.europa.esig.dss.tsl.service.TSLParser.getService(TSLParser.java:339)
	at eu.europa.esig.dss.tsl.service.TSLParser.getServices(TSLParser.java:327)
	at eu.europa.esig.dss.tsl.service.TSLParser.getServiceProvider(TSLParser.java:317)
	at eu.europa.esig.dss.tsl.service.TSLParser.getServiceProviders(TSLParser.java:303)
	at eu.europa.esig.dss.tsl.service.TSLParser.getTslModel(TSLParser.java:152)
	at eu.europa.esig.dss.tsl.service.TSLParser.call(TSLParser.java:137)
	... 5 more
```

This is the code, that causes it:

```java
Configuration conf = new Configuration();
conf.setTspSource("http://demo.sk.ee/tsa/");
Container container = ContainerBuilder.aContainer()
	.withDataFile("/path/to/my/test/file.xml", "text/xml")
	.withConfiguration(conf).build();

String privateKeyPath = "/path/to/my/private/key.p12";
char[] password = "my-password".toCharArray();
PKCS12SignatureToken signatureToken = new PKCS12SignatureToken(privateKeyPath, password);
SignatureBuilder.aSignature(container).withSignatureToken(signatureToken).invokeSigning();
```

The exception happens on the last line.

The error seems to be caused by the Portugal government giving bad data to the library. The error itself isn't a problem, but the fact that it prints a stack trace into my logs is a problem. If I wish to ignore this error with log4j, I have to ignore the class `TSLValidationJob.java`, but then I may be ignoring other important errors.

The fix seems to be for the Portugal government to fix their mistake, but I figured going through the proper channels is more effective.

**Comment by raigoinabox** [comment]
I thought about this and decided to leave it how it is. I discovered that the error is shown only once during the whole application run time, which is a lot smaller hassle, I'd rather not reduce the trusted territories and with my small testing, setting TRUSTED_TERRITORIES: EE didn't work anyhow.

**Comment by naare** [comment]
I'll close the ticket as the original problem has "resolved".
Configuring the TRUSTED_TERRITORIES should work as I tested it. If you need assistance on that please open another ticket.
